### ISSUE TITLE
What's the preferred linux package name for OSM Express?

### ISSUE THREAD

**Issue by das-g** [protomaps/OSMExpress#19]
Cool project! I'm considering to package OSM Express for [NixOS](https://nixos.org) / [nixpkgs](https://github.com/NixOS/nixpkgs).

Package names in nixpkgs:
- must be lowercase
- cannot contain spaces
- may be hyphenated

(See section [Package naming](https://nixos.org/nixpkgs/manual/#sec-package-naming) in the Nixpkgs manual.)

Given these constraints, what should the package name be?

- `osmexpress`?
- `osm-express`?
- `osmx`, like your PyPI package for the python lib?

**Comment by bdon** [comment]
Hmm, my inclination is that `osmexpress` feels less arcane than just `osmx`. If we use that, maybe we should change the command line tool name and python package as well?

**Comment by das-g** [comment]
> Hmm, my inclination is that `osmexpress` feels less arcane than just `osmx`.

The long forms are more explicit about what the name means, yeah. As you didn't mention the variant with a hyphen, do you prefer `osmexpress` over `osm-express`? "osm" already being an abbreviation, I find the latter easier to decipher.

> If we use that, maybe we should change the command line tool name and python package as well?

While it's quite common to have command names and package names match (especially with packages that provide exactly one command), command names that deviate from the package name aren't unheard of. Take `mercurial`, whose main command is `hg`. Because a command is usually typed much more often than a package name, brevity can be a valid concern.

I don't know how established the `osmx` Python package already is. If it's in wide use already, you might wish to keep its name.

**Comment by bdon** [comment]
python import statements can't have `-`, so avoiding `_` or `-` seems more flexible in general. I don't believe there are many users of the python package right now, so that shouldn't factor into changing the name.

**Comment by brawer** [comment]
Curious, has this been resolved?

**Comment by bdon** [comment]
I don't believe so, no results on https://search.nixos.org/packages and I haven't ever packaged this project into a package manager.

I would still prefer `osmexpress` to disambiguate from the similarly named and popular OSMnx (https://osmnx.readthedocs.io) project.

**Comment by das-g** [comment]
Sorry, I kinda forgot about this. I don't know whether I'll have time to attempt it any time soon, so if someone else feels like packaging it for NixPkgs, feel free to go forward.
